albogdan/skule-vote
# Create Models

## Overview

- Resolves #1
- Creates models for `ElectionSession`, `Election`, `Candidate`, `Voter`, `Ballot`, `Eligibility`, `Message`.
- Creates Admin classes that allow the models to be seen on the Django Admin site.
- Creates an initial database migration for the models created.


## Unit Tests Created

- None created. Only models created.


## Steps to QA

- Once you pull make sure you run `python manage.py migrate` to apply the database migrations
- Go to the admin site `http://127.0.0.1:8000/admin` and do the following.
    - Ensure all 7 models are visible under the `Backend` group
    - Go through all the tabs and ensure the help text is okay (help text in this case is for example when you add a new Election Session under the "Election session name" text box there is `Name your Election Session (i.e., Fall 2021).`. This help text occurs for all the modules except for `Voter` and `Ballot`. 
        - Note that for the `Eligibility` table the help text is just the text box beside the check mark. Go through the code for this one and ensure that the help text matches the discipline in the variable name.
    - Click on `Election Session` and create one.
    - Click on `Election` and create one. Ensure you see the `Election Session` created previously in the dropdown.
    - Click on `Candidates` and create one. Ensure you see the `Election` created previously in the dropdown. You only need to fill up to and including the `Preamble` section. Try saving. You should get no errors even though all the boxes below that are blank. There will be more to clarify this in a future PR which flushes out admin site details.
    - Click on `Candidates` and create one. Ensure you see the `Election` created previously in the dropdown. Fill in whichever checkboxes you want and try saving. You should get an error because the `Status eligible` at the bottom was not filled in (this is a required field. The default for all of the checkbox fields is False.
    - Click on `Messages` and create one. Ensure you see the `Election Session` created previously in the dropdown. Enter a message and save. The message is active and un-hideable by default.

- Ensure all the fields we discussed in the spreadsheet are in the proper models and all of them exist and have the proper options (for the ones with select options)
- Make sure the help text is to your liking. Let me know if I should change anything. 
- I'm kind of iffy on the _str_ fcn for `Message` - any ideas on how to improve it?
